### ISSUE TITLE
tab completing 'vele' results in '_velero'

### ISSUE THREAD

**Issue by Hr46ph** [Homebrew/homebrew-core#159321]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
I installed velero using homebrew but when I want to tab-complete the command it completes to "_velero". This file is the script for velero tab completion.

It could be my environment, I am not sure but I searched to troubleshoot this and I can't find why or how this would happen.

I have several other cli utilities like talosctl helm and kubectl that source the same way (ie, `source <(command completion zsh)`.
```

### Verification

- [X] My "`brew doctor` output" says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.
- [X] I searched for recent similar issues at https://github.com/Homebrew/homebrew-core/issues?q=is%3Aissue and found no duplicates.

### What were you trying to do (and why)?

I simply want to tab-complete a command, velero.

### What happened (include all command output)?

Tab completing "vele" completes to "_velero", the completion script for that command.

### What did you expect to happen?

tab-completing to "velero", without the understore.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
brew install velero
type "vele" and hit TAB
```
Brew config output:
```
brew config                                                                                                                                                                    ─╯
HOMEBREW_VERSION: 4.2.3
ORIGIN: https://github.com/Homebrew/brew
HEAD: b3751bca8c4a3c76107d5aa3b75b81db93cf4bb6
Last commit: 15 hours ago
Core tap JSON: 08 Jan 14:53 UTC
Core cask tap JSON: 08 Jan 14:53 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.2.1-arm64
CLT: 15.1.0.0.1.1700200546
Xcode: N/A
Rosetta 2: false
```

**Comment by SMillerDev** [comment]
What shell are you using?

**Comment by Hr46ph** [comment]
Ugh who says computes don't have a sense of humor. It lets me struggle for a while and then when I post a bug report, it suddenly behaves! 

I have cleared up the brew doctor issues. 3 unlinked kegs, I uninstalled python-idna, linked python-lxml --overwrite (after checking) and uninstalled httpx. I also uninstalled lua@5.3, openssl@1.1 and packer (replaced with hashicorp/tap/packer). Doctor stopped complaining. Restarted the shell several times in between those actions.

vele+TAB now suddenly completes without the underscore.

Beats me really, I don't see the above actions could have interfered. I had already removed coreutils and util-linux from the path since that might lead to issues and that wasn't it. I checked aliases and other paths in my path statement.

Anyway, apologies for the noise, but its working fine now although I have no idea what solved it.
